src/oidc/configuration.ts

```diff
@@ -18,23 +21,54 @@ export default function createOIDCConfig(model: Model, oidcConfig: Config['oidc'
   return {
     adapter,
     findAccount: async (ctx, id) => {
-      const [groups, username] = await model.pgDo(async tr => {
+      const [profile, email, groups] = await model.pgDo(async tr => {
+        // get name and username
+        const userResult = await model.users.getByUserIdx(tr, Number(id));
+        if (!userResult.name || !userResult.username) {
+          throw new Error('name or username empty');
+        }
+
+        // ~84: YYXX-NNNN, 85 ~ 99: YYXXX-NNN, 00 ~: YYYY-NNNNN
+        // get student id, hard-coded by ataching '19' and sorting
+        const sidResult = await model.users.getStudentNumbersByUserIdx(tr, Number(id));
+        if (sidResult.length === 0) {
+          throw new Error('no student id');
+        }
+        const primarySid = sidResult
+          .map(sid => ({
+            sid,
+            year: Number(sid.length === 9 ? `19${sid.substring(0, 2)}` : sid.substring(0, 4)),
+          }))
+          .sort((a, b) => b.year - a.year)
+          .map(({ sid }) => sid)[0];
+
+        const profile = {
+          name: userResult.name,
+          username: userResult.username,
+          student_id: sidResult[0],
```

Comment:
```suggestion
          student_id: primarySid,
```

`primarySid` 안 쓴다고 밑줄 쳐주지 않나요

Reply:
쳐주는데 못봄